backend/layers/business/business.py

```diff
@@ -972,6 +972,18 @@ def publish_collection_version(
             if canonical_datasets != version_datasets:
                 has_dataset_revisions = True
 
+            # Update publisher metadata if there are updates in Crossref that have occurred since last publish.
+            doi_link = next((link for link in version.metadata.links if link.type == "DOI"), None)
+            if doi_link:
+                # Don't track/raise errors here; just keep the existing publisher metadata.
+                publisher_metadata, _, deposited_at_timestamp = self._get_publisher_metadata(doi_link.uri, [])
```

Comment:
to update DOI if there is a new DOI, get that 2nd return value (doi_curie_from_crossref)

only if `f"https://doi.org/{doi_curie_from_crossref}" !=  doi_link.uri` (that is, we passed in a pre-print DOI but there is now a published DOI available that we've fetched in _get_publisher_metadata) then--build CollectionMetadataUpdate object with just the 'links' field (pass in entire list of links, replacing `doi_link.uri` in the list with `f"https://doi.org/{doi_curie_from_crossref}`). Pass into update_collection_version with `apply_doi_links=True. `. Do NOT run
self.database_provider.finalize_collection_version if this path is invoked, instead raise a CollectionPublishException explaining that the DOI was updated and artifact updates are required and to retry after updates. User will need to attempt re-publish once the artifact updates invoked by update_collection_version complete.
 
Otherwise, if doi_curie_from_crossref is not new, continue with the logic that's been added here (looks good!)

Reply:
Oh, and if DOI is updated, update_collection_version will also take care of updating the publisher_metadata. But the logic here looks good for the scenario where DOI is not updated but there is new publisher metadata